packages/hathor-rpc-handler/src/rpcMethods/sendNanoContractTx.ts

```diff
@@ -24,11 +24,32 @@ import {
 import { PromptRejectedError, SendNanoContractTxError, InvalidParamsError } from '../errors';
 import { NanoContractAction } from '@hathor/wallet-lib/lib/nano_contracts/types';
 
+export type NanoContractActionWithStringAmount = Omit<NanoContractAction, 'amount'> & {
+  amount: string,
+}
+
 const sendNanoContractSchema = z.object({
   method: z.string().min(1),
   blueprint_id: z.string().nullish(),
   nc_id: z.string().nullish(),
-  actions: z.array(z.custom<NanoContractAction>()),
+  actions: z.array(z.custom<NanoContractActionWithStringAmount>()
+    .refine(action => {
+      try {
+        // Check if amount is a valid number string that can be converted to BigInt
+        const bigIntVal = BigInt(action.amount);
+        return bigIntVal >= BigInt(0); // Ensure non-negative value
+      } catch (e) {
+        return false;
+      }
+    }, {
+      message: "Amount must be a valid non-negative number string"
+    })
+  ).transform(actions => 
+    actions.map(action => ({
+      ...action,
+      amount: BigInt(action.amount)
+    }))
+  ),
```

Comment:
Maybe we should use the coercible bigint since it is less verbose.

Reply:
Done, thanks